services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainController.java

```diff
@@ -769,7 +769,31 @@ protected void initKzg() {
                     return DEFAULT_KZG_PRECOMPUTE;
                   });
 
-      kzg.loadTrustedSetup(trustedSetupFile, kzgPrecompute);
+      // Run loadTrustedSetup on a dedicated thread with a larger stack size.
+      final long stackSize = 8 * 1024 * 1024; // 8MB
+      final AtomicReference<Throwable> loadException = new AtomicReference<>();
+      final Thread loaderThread =
+          new Thread(
+              null,
+              () -> {
+                try {
+                  kzg.loadTrustedSetup(trustedSetupFile, kzgPrecompute);
+                } catch (final Throwable t) {
+                  loadException.set(t);
+                }
+              },
+              "kzg-loader",
+              stackSize);
+      loaderThread.start();
+      try {
+        loaderThread.join();
```

Comment:
thoughts on maybe some kind of sane timeout here? even if its a longish one, just not unbound...